File: MultiLevel.java
class Base
{
    public int A, B;
    public Base()
    {
        System.out.println("Inside Base constructor");
    }
    public void Fun()
    {
        System.out.println("Inside Fun of Base");
    }
}

class Derived extends Base  // class Derived : public Base
{
    public int X, Y;
    public Derived()
    {
        System.out.println("Inside Derived constructor");
    }
    public void Gun()
    {
        System.out.println("Inside Gun of Derived");
    }
}

class DerivedX extends Derived
{
    public int L, M;
    public DerivedX()
    {
        System.out.println("Inside DerivedX constructor");
    }
    public void Sun()
    {
        System.out.println("Inside Sun of DerivedX");
    }
}

class MultiLevel
{
    public static void main(String A[])
    {
        System.out.println("Inside main");

        DerivedX dobj = new DerivedX();
        dobj.Fun();
        dobj.Gun();
        dobj.Sun();
    }
}

File: Multilevel.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        int i,j;
        Base()
        {
            cout<<"Base constructor\n";
        }
        ~Base()
        {
            cout<<"Base destructor\n";
        }
};

class Derived : public Base
{
    public:
        int k;
        Derived()
        {
            cout<<"Derived constructor\n";
        }
        ~Derived()
        {
            cout<<"Derived destructor\n";
        }
};

class DerivedX : public Derived
{
    public:
        int p,q;
        DerivedX()
        {
            cout<<"DerivedX constructor\n";
        }
        ~DerivedX()
        {
            cout<<"DerivedX destructor\n";
        }
};

int main()
{
    DerivedX dobj;

    cout<<"Size of Base is : "<<sizeof(Base)<<"\n";             // 8
    cout<<"Size of Derived is : "<<sizeof(Derived)<<"\n";       // 12
    cout<<"Size of DerivedX is : "<<sizeof(DerivedX)<<"\n";     // 20

    return 0;
}